Marlin/src/lcd/menu/menu.cpp

```diff
@@ -228,21 +229,73 @@ void MarlinUI::goto_screen(screenFunc_t screen, const uint16_t encoder/*=0*/, co
 ///////////// Manual Movement //////////////
 ////////////////////////////////////////////
 
+// Helper vars for custom msg pointers
+static FSTR_P message_ptrs[2];
+static char item_line_buf[LCD_WIDTH+1];
+
 //
 // Display a "synchronize" screen with a custom message until
 // all moves are finished. Go back to calling screen when done.
 //
 void MarlinUI::synchronize(FSTR_P const fmsg/*=nullptr*/) {
-  static FSTR_P sync_message = fmsg ?: GET_TEXT_F(MSG_MOVING);
+  message_ptrs[0] = fmsg ?: GET_TEXT_F(MSG_MOVING);
   push_current_screen();
   goto_screen([]{
-    if (should_draw()) MenuItem_static::draw(LCD_HEIGHT >= 4, sync_message);
+    if (should_draw()) MenuItem_static::draw(LCD_HEIGHT / 2 - 1, message_ptrs[0]);
   });
   defer_status_screen();
   planner.synchronize(); // idle() is called until moves complete
   goto_previous_screen_no_defer();
 }
 
+/**
+ * Displays screen with custom message.
+ *   Message is automatically wrapped to fit screen width.
+ *   Line breaks can be forced by "\n"
+ *   Takes up to 2 FSTR_P (str1, str2) that are seamlessly joined
+ */
+void MarlinUI::goto_message_screen(FSTR_P const str1/*=nullptr*/, FSTR_P const str2/*=nullptr*/) {
+
+  static uint8_t ptr_count;
+  ptr_count = 0;
+
+  if (str1 != nullptr) message_ptrs[ptr_count++] = str1;
+  if (str2 != nullptr) message_ptrs[ptr_count++] = str2;
+
+  push_current_screen();
+  goto_screen([]{
+    if (should_draw()) {
+
+      uint8_t item_line_pos = 0;
+      uint8_t line_to_draw = 0;
+
+      for (uint8_t p = 0; p < ptr_count; p++) {
+
+        const char * str = FTOP(message_ptrs[p]);
+        const uint8_t str_len = strlen_P(str);
+
+        for (uint8_t i = 0; i <= str_len; i++) {
+          const char c = str[i];
+
+          if (c == '\n' || item_line_pos == LCD_WIDTH || (c == '\0' && p == ptr_count-1)) {
+            item_line_buf[item_line_pos] = '\0';
+            const uint8_t lcd_line = LCD_HEIGHT / 2 - 1 + line_to_draw;
+            if (lcd_line < LCD_HEIGHT) { // implementation limited
+              MenuItem_static::draw(lcd_line, FPSTR(item_line_buf));
```

Comment:
It's not possible to coerce the SRAM string `item_line_buf` into a PROGMEM string. This will not work on AVR, and will just display garbage.

Reply:
This message screen concept is basically the same (with no buttons) as `MenuItem_confirm::draw_select_screen` which uses the existing method `MarlinUI::draw_select_screen_prompt` to draw the provided `FSTR_P`. It uses `wrap_string_P(col, row, FTOP(fstr), true)` to split up PROGMEM strings that are wider than the screen and properly handles UTF-8.

Reply:
I fixed the SRAM/PROGMEM issue by using the `MenuItem_static::draw` with `nullptr` for `fstr` and `item_line_buf` as `vstr`.

Using `draw_select_screen` is a great idea with one caveat 🙂, it is not implemented for HAS_GRAPHICAL_TFT, because that uses completely separate select screen implementation. So for now I will be using my function so that the Tramming Wizard is done and I will consolidate those methods in the future.